pyomo/contrib/mindtpy/config_options.py

```diff
@@ -575,4 +575,23 @@ def check_config(config):
                 'The threads parameter is corrected to 1 since incumbent callback conflicts with multi-threads mode.')
     if config.solution_pool:
         if config.mip_solver not in {'cplex_persistent', 'gurobi_persistent'}:
+            if config.mip_solver in {'appsi_cplex', 'appsi_gurobi'}:
+                config.logger.info("Solution pool does not support APPSI solver yet.")
             config.mip_solver = 'cplex_persistent'
+    if config.calculate_dual:
+        if config.mip_solver == 'appsi_cplex':
+            config.logger.info("APPSI-Cplex cannot get duals for mixed-integer problems"
+                            "mip_solver will be changed to Cplex")
+            config.mip_solver = 'cplex'
+        if config.mip_regularization_solver == 'appsi_cplex':
+            config.logger.info("APPSI-Cplex cannot get duals for mixed-integer problems"
+                            "mip_solver will be changed to Cplex")
+            config.mip_regularization_solver = 'cplex'
+        if config.mip_regularization_solver == 'appsi_gurobi':
+            config.logger.info("APPSI-GUROBI cannot get duals for mixed-integer problems"
+                            "mip_solver will be changed to GUROBI")
+            config.mip_regularization_solver = 'gurobi'
+        if config.mip_regularization_solver == 'appsi_gurobi':
+            config.logger.info("APPSI-GUROBI cannot get duals for mixed-integer problems"
+                            "mip_solver will be changed to GUROBI")
+            config.mip_regularization_solver = 'gurobi'
```

Comment:
Does this work? I did not think Gurobi provided duals for mixed-integer problems regardless of the Pyomo interface.

Reply:
You are right. Fixed.